### PR TITLE
Add new keyboard - TouhouPad

### PR DESCRIPTION
**Pull request by Pimentoso**
## Description

Add support for the TouhouPad mechanical keypad.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/pimentoso/touhoupad/readme.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# TouhouPad
+
+![TouhouPad](https://imgur.com/a/jcuEr8q)
```
Comment (fauxpark) [inline keyboards/pimentoso/touhoupad/readme.md:3]:
```suggestion
![TouhouPad](https://i.imgur.com/JeEqeEs.png)
```

File: keyboards/pimentoso/touhoupad/readme.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+A 10 key gamepad specifically for playing Touhou danmaku games.
+
+You can find building instructions and gerber files to print the circuit board on the [project page.](https://github.com/Pimentoso/TouhouPad_v2)
+
+* Keyboard Maintainer: [Michele Ferri](https://github.com/Pimentoso)
+* Hardware Supported: Paddino02 PCB rev1, rev2 + Arduino Pro Micro
+* Hardware Availability: the PCB is open source and can be freely printed at websites such as [JLCPCB](https://jlcpcb.com/)
+### To flash (after setting up your build environment):
+
+`make pimentoso/paddino02/rev2/right:default:flash`
```
Comment (fauxpark) [inline keyboards/pimentoso/touhoupad/readme.md:14]:
```suggestion

Make example for this keyboard (after setting up your build environment):

    make pimentoso/touhoupad:default
```

File: keyboards/pimentoso/touhoupad/keymaps/default/config.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/* Copyright 2021 Michele Ferri <zomgsako@gmail.com>
```
Comment (fauxpark) [inline keyboards/pimentoso/touhoupad/keymaps/default/config.h:1]: This file can be deleted.

File: keyboards/pimentoso/touhoupad/rules.mk [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/pimentoso/touhoupad/rules.mk:11]:
```suggestion
```

File: keyboards/pimentoso/touhoupad/rules.mk [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/pimentoso/touhoupad/rules.mk:31]:
```suggestion
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/pimentoso/touhoupad/config.h [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+#define RGB_DI_PIN F4
+#ifdef RGB_DI_PIN
+  #define RGBLED_NUM 12
+  #define RGBLIGHT_HUE_STEP 8
+  #define RGBLIGHT_SAT_STEP 8
+  #define RGBLIGHT_VAL_STEP 8
+  #define RGBLIGHT_LIMIT_VAL 255 /* The maximum brightness level */
+  #define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+  #define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/pimentoso/touhoupad/config.h:43]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**